include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -30,7 +30,7 @@ def FIRRTLType : FIRRTLDialectType<CPred<"$_self.isa<FIRRTLType>()">,
   }]>;
 
 def FIRRTLBaseType : FIRRTLDialectType<
-  CPred<"$_self.isa<FIRRTLBaseType>()">,
+  CPred<"isa<FIRRTLBaseType>($_self)">,
```

Comment:
Should this be `firrtl::type_isa`?

Reply:
Ah, looks like while visiting also moving `mlir::Type::isa<T>` -> `isa<T>` "free" function at least throughout this file.

Reply:
Yes, we need tyep_isa only for subtypes of FIRRTLBaseType.